### PR TITLE
Snake LL

### PR DESCRIPTION
**Pull request by Quaqqer**
Snake game written with linked list implementation, has a crash when accessing `board.is_running` after `board.is_running` for some reason.

### CODE REVIEW

File: jordgubb/snake_ll.sy [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    Left,
+    Right,
+    None,
+end
+
+dir_tup :: fn dir: Direction -> (int, int) do
+    case dir do
+        Up do
+            ret ( 0, -1)
+        end
```
Comment (FredTheDino) [inline jordgubb/snake_ll.sy:40]: Fun fact - this is legal:
```suggestion
        Up ret ( 0, -1)
```

File: jordgubb/snake_ll.sy [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+use input
+
+MaybeSegment :: enum
+    Just Segment,
+    None,
+end
+
+Segment :: blob {
+    position: (int, int),
+    child: *,
```
Comment (FredTheDino) [inline jordgubb/snake_ll.sy:13]: :(

**Review by FredTheDino — COMMENTED** [review]
Looks sick! Gonna need to play it! :D

**Review by FredTheDino — APPROVED** [review]

File: jordgubb/snake_ll.sy (hunk 2 above)
Comment (EMattfolk) [reply jordgubb/snake_ll.sy:13]: It is not optimal. I don't know if it is worth fixing before the jam...

**Review by EMattfolk — APPROVED** [review]

**Review by sornas — APPROVED** [review]
game of the year

**Pull request merged**